c10/core/DispatchKey.h

```diff
@@ -165,9 +167,11 @@ enum class DispatchKey : uint16_t {
   // If any of these backends ever need to customize, e.g., Autograd, then we'll
   // need to add a DispatchKey::*Bit for them.
 
+  // TODO: put this in BackendComponents
```

Comment:
Do you think we want to put these extra backends into `BackendComponent` in the long term?

Pro: It simplifies our model; we don't have to worry about what "fake backends" mean w.r.t the dispatcher

Con: We're providing extra flexibility preemptively, since none of these backends have asked for the ability to override autograd/sparse/quantized/etc. Also, it'll blow up the size of our dispatch table some more (but maybe this is ok and we just need to measure the perf impact).